src/backend/utils/misc/guc_tables.c

```diff
@@ -3650,6 +3650,17 @@ struct config_int ConfigureNamesInt[] =
 		NULL, NULL, NULL
 	},
 
+	{
+		{"neon_pgstat_file_size_limit", PGC_SIGHUP, STATS_CUMULATIVE,
+			gettext_noop("Maximal size of pg_stat file saved in Neon storage."),
+			NULL,
+			GUC_UNIT_KB
+		},
+		&neon_pgstat_file_size_limit,
+		1024, 100, 1048576,
```

Comment:
Setting this to 1048576 will cause the palloc in wallog_file_descriptor to fail if the file has just the right size, because 1048576*1024 > MaxAllocSize. I think you'll run into limits in the WAL record size too. Maybe dial down the max value to `1000000` or something.

I'd suggest making the minimum value 0, to have a clean way to cause the limit to always be exceeded.

Reply:
Done